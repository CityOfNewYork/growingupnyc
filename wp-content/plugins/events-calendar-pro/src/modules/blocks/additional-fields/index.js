export { default as Text } from './text/container';
export { default as URL } from './url/container';
export { default as TextArea } from './text-area/container';
export { default as Dropdown } from './dropdown/container';
export { default as Radio } from './radio/container';
export { default as Checkbox } from './checkbox/container';
