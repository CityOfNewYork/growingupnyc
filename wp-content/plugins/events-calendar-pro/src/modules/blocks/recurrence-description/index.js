/**
 * External dependencies
 */
import { DateTime } from '@moderntribe/events/icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import EventDescription from './container';
import './style.pcss';

/**
 * Module Code
 */

export default {
	id: 'event-pro-recurrence-description',
	title: __( 'Recurring Description', 'events-gutenberg' ),
	description: __(
		'Recurrence description',
		'events-gutenberg'
	),
	icon: <DateTime />,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	parent: [ 'tribe/event-datetime-content' ],

	supports: {
		html: false,
	},

	attributes: {
		description: {
			type: 'string',
			source: 'meta',
			meta: '_tribe_blocks_recurrence_description',
		},
	},

	edit: EventDescription,

	save( props ) {
		return null;
	},
};
