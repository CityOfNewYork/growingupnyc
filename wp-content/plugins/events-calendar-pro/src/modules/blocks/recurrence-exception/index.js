/**
 * External dependencies
 */
import { DateTime } from '@moderntribe/events/icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import EventException from './container';

/**
 * Module Code
 */

export default {
	id: 'event-pro-recurrence-exception',
	title: __( 'Exception', 'tribe-events-calendar-pro' ),
	description: __(
		'Add exceptions to your event.',
		'tribe-events-calendar-pro',
	),
	icon: <DateTime />,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	parent: [ 'tribe/event-pro-recurrence' ],

	supports: {
		html: false,
	},

	edit: EventException,

	save( props ) {
		return null;
	},
};
