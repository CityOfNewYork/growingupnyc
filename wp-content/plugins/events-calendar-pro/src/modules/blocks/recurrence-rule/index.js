/**
 * External dependencies
 */
import { DateTime } from '@moderntribe/events/icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import EventRecurring from './container';

/**
 * Module Code
 */

export default {
	id: 'event-pro-recurrence-rule',
	title: __( 'Rules', 'tribe-events-calendar-pro' ),
	description: __(
		'Add recurrence to your event.',
		'tribe-events-calendar-pro'
	),
	icon: <DateTime />,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	parent: [ 'tribe/event-pro-recurrence' ],

	supports: {
		html: false,
	},

	edit: EventRecurring,

	save( props ) {
		return null;
	},
};
