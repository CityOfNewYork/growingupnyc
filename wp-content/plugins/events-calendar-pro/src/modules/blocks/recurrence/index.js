/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { DateTime } from '@moderntribe/events/icons';

/**
 * Internal dependencies
 */
import EventRecurring from './container';

/**
 * Module Code
 */

export default {
	id: 'event-pro-recurrence',
	title: __( 'Recurring', 'events-gutenberg' ),
	description: __(
		'Entry for recurrence',
		'events-gutenberg'
	),
	icon: <DateTime />,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	parent: [ 'tribe/event-datetime' ],

	supports: {
		html: false,
	},

	attributes: {
		exceptions: {
			type: 'string',
			source: 'meta',
			meta: '_tribe_blocks_recurrence_exclusions',
		},
		rules: {
			type: 'string',
			source: 'meta',
			meta: '_tribe_blocks_recurrence_rules',
		},
	},

	edit: EventRecurring,

	save( props ) {
		return null;
	},
};
