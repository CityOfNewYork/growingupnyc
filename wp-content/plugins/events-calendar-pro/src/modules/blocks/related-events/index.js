/**
 * External dependencies
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import RelatedEvents from './container';
import { Sharing } from '@moderntribe/events/icons';

/**
 * Module Code
 */
export default {
	id: 'related-events',
	title: __( 'Related Events', 'tribe-events-calendar-pro' ),
	description: __(
		'Show other events with the same event categories and/or tags.',
		'tribe-events-calendar-pro'
	),
	icon: <Sharing/>,
	category: 'tribe-events',
	keywords: [ 'event', 'events-gutenberg', 'tribe' ],

	supports: {
		html: false,
	},

	attributes: {
		title: {
			type: 'html',
			default: __( 'Related Events', 'tribe-events-calendar-pro' ),
		},
		displayImages: {
			type: 'boolean',
			default: true,
		},
	},

	edit: RelatedEvents,
	save: () => null,
};

