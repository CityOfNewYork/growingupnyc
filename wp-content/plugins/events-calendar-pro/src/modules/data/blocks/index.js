/**
 * Internal dependencies
 */
import * as constants from './constants';
import * as recurring from './recurring';
import reducer from './reducer';

export default reducer;
export { constants };
export { recurring };
