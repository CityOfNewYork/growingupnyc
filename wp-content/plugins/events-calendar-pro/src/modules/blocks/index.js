/**
 * External Dependencies
 */
import { registerBlockType } from '@wordpress/blocks';

/**
 * Internal Dependencies
 */
import recurrence from '@moderntribe/events-pro/blocks/recurrence';
import recurrenceRule from '@moderntribe/events-pro/blocks/recurrence-rule';
import recurrenceException from '@moderntribe/events-pro/blocks/recurrence-exception';
import recurrenceDescription from '@moderntribe/events-pro/blocks/recurrence-description';
import RelatedEvents from '@moderntribe/events-pro/blocks/related-events';
import { addAdditionalFields } from '@moderntribe/events-pro/blocks/additional-fields/utils';
import { initStore } from '@moderntribe/events-pro/data';

const blocks = addAdditionalFields( [
	recurrence,
	recurrenceRule,
	recurrenceException,
	recurrenceDescription,
	RelatedEvents,
] );

blocks.forEach( block => {
	const blockName = `tribe/${ block.id }`;
	registerBlockType( blockName, block );
} );

// Initialize AFTER blocks are registered
// to avoid plugin shown as available in reducer
// but not having block available for use
initStore();

export default blocks;
