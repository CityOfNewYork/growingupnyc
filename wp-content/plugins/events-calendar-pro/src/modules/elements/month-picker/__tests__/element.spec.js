import React from 'react';
import renderer from 'react-test-renderer';
import MonthPicker from '../element';
import { options } from '@moderntribe/events-pro/data/blocks/recurring';

describe( 'MonthPicker', () => {
	test( 'should render component', () => {
		const component = renderer.create(
			<MonthPicker
				className="test-class"
				months={ [] }
				onMonthClick={ jest.fn() }
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );

	test( 'should render month tags', () => {
		const component = renderer.create(
			<MonthPicker
				className="test-class"
				months={ [
					options.MONTHS_OF_THE_YEAR_OPTIONS[6],
					options.MONTHS_OF_THE_YEAR_OPTIONS[8],
				] }
				onMonthClick={ jest.fn() }
			/>
		);
		expect( component.toJSON() ).toMatchSnapshot();
	} );
} );
