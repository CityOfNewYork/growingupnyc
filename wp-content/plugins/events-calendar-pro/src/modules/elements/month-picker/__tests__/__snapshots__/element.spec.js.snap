// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonthPicker should render component 1`] = `
<div
  className="tribe-editor__month-picker test-class"
>
  <button
    className="tribe-editor__month-picker__add"
    onClick={[Function]}
  >
    <span
      className="tribe-editor__month-picker__add-icon"
    >
      +
    </span>
    Add
  </button>
</div>
`;

exports[`MonthPicker should render month tags 1`] = `
<div
  className="tribe-editor__month-picker test-class"
>
  <button
    className="tribe-editor__month-tag tribe-editor__month-picker__month-tag"
  >
    <<svg>icon mock</svg> />
    <span
      className="tribe-editor__month-tag__remove"
    >
      Remove
    </span>
    Jul
  </button>
  <button
    className="tribe-editor__month-tag tribe-editor__month-picker__month-tag"
  >
    <<svg>icon mock</svg> />
    <span
      className="tribe-editor__month-tag__remove"
    >
      Remove
    </span>
    Sep
  </button>
  <button
    className="tribe-editor__month-picker__add"
    onClick={[Function]}
  >
    <span
      className="tribe-editor__month-picker__add-icon"
    >
      +
    </span>
    Add
  </button>
</div>
`;
